erts/emulator/beam/erl_arith.c

```diff
@@ -461,6 +461,70 @@ erts_mixed_plus(Process* p, Eterm arg1, Eterm arg2)
     }
 }
 
+Eterm
+erts_unary_minus(Process* p, Eterm arg)
```

Comment:
LGTM. I would just ask for a comment above this function head explaining that the reason it exists is that "-X" is not the same as "0-X" when X is floating-point zero.

Reply:
Done!